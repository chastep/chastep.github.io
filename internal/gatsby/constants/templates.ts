import path from "path";

const templates = Object.freeze({
  indexTemplate: path.resolve("./src/templates/index-template.tsx"),
  categoryTemplate: path.resolve("./src/templates/category-template.tsx"),
  notFoundTemplate: path.resolve("./src/templates/not-found-template.tsx"),
  categoriesTemplate: path.resolve("./src/templates/categories-template.tsx"),
  tagsTemplate: path.resolve("./src/templates/tags-template.tsx"),
  pageTemplate: path.resolve("./src/templates/page-template.tsx"),
  postTemplate: path.resolve("./src/templates/post-template.tsx"),
  tagTemplate: path.resolve("./src/templates/tag-template.tsx"),
});

export { templates };
