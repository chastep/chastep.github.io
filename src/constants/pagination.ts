const pagination = {
  prevPage: "← PREV",
  nextPage: "→ NEXT",
};

export { pagination }
