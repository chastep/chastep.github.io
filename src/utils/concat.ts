const concat = (...args: string[]): string => args.join("");

export { concat };
