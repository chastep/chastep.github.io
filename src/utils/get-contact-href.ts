const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    mastodon: `${contact}`,
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    x: `https://www.x.com/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    bluesky: `https://bsky.app/profile/${contact}.bsky.social`,
  };

  return hrefs[name] ?? contact;
};

export { getContactHref };
