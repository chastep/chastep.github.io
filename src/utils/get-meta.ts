const getMeta = (container: HTMLElement, key: string) => {
  const meta = container.getElementsByTagName("meta");

  for (let i = 0; i < meta.length; i += 1) {
    if (
      key === meta[i].getAttribute("name") ||
      key === meta[i].getAttribute("property")
    ) {
      return meta[i].getAttribute("content");
    }
  }

  return "";
};

export { getMeta };
