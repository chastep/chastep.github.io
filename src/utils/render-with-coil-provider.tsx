import React, { type ReactElement, type ReactNode, type FC } from "react";
import { CoilProvider } from "@alxshelepenok/diesel";
import { render, type RenderOptions } from "@testing-library/react";
import renderer, { type ReactTestRenderer, type TestRendererOptions } from "react-test-renderer";

interface WithCoilProviderProps {
  children: ReactNode;
}

const WithCoilProvider: FC<WithCoilProviderProps> = ({ children }) => (
  <CoilProvider>{children}</CoilProvider>
);

const renderWithCoilProvider = (
  ui: ReactElement,
  options?: RenderOptions,
) => render(ui, { wrapper: WithCoilProvider, ...options });

export const createSnapshotsRenderer = (
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer =>
  renderer.create(<CoilProvider>{nextElement}</CoilProvider>, options);

export { renderWithCoilProvider };
