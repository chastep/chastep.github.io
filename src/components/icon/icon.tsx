import React, { type FC } from "react";

import { icons } from "@/constants/icons";

import * as styles from "./icon.module.scss";

interface IconProps {
  name: keyof typeof icons;
  icon: {
    path?: string;
    viewBox?: string;
  };
}

const Icon: FC<IconProps> = ({ name, icon }) => (
  <svg className={styles.icon} viewBox={icon.viewBox}>
    <title>{name}</title>
    <path d={icon.path} />
  </svg>
);

export { Icon };
