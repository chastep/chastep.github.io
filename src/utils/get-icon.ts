import { icons } from "@/constants/icons";

const getIcon = (name: keyof typeof icons) => icons[name] || {};

export { getIcon };
