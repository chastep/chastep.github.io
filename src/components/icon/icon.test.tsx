import React from "react";
import { test, describe, expect } from "bun:test";

import { Icon } from "@/components/icon/icon";
import { icons } from "@/constants/icons";
import { getIcon } from "@/utils/get-icon";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("Icon", () => {
  test("renders correctly", () => {
    const [x] = Object.keys(icons) as Array<keyof typeof icons>;
    const props = { name: x, icon: getIcon(x) };
    const tree = createSnapshotsRenderer(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with custom size", () => {
    const [x] = Object.keys(icons) as Array<keyof typeof icons>;
    const props = { name: x, icon: getIcon(x), size: 32 };
    const tree = createSnapshotsRenderer(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with custom color", () => {
    const [x] = Object.keys(icons) as Array<keyof typeof icons>;
    const props = { name: x, icon: getIcon(x), color: "red" };
    const tree = createSnapshotsRenderer(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with custom size and color", () => {
    const [x] = Object.keys(icons) as Array<keyof typeof icons>;
    const props = {
      name: x,
      icon: getIcon(x),
      size: 32,
      color: "red",
    };
    const tree = createSnapshotsRenderer(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
