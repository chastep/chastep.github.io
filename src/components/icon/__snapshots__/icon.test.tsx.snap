// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "x",
      ],
      "props": {},
      "type": "title",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "d": "M14.095479,10.316482L22.286354,1h-1.940718l-7.115352,8.087682L7.551414,1H1l8.589488,12.231093L1,23h1.940717  l7.509372-8.542861L16.448587,23H23L14.095479,10.316482z M11.436522,13.338465l-0.871624-1.218704l-6.924311-9.68815h2.981339  l5.58978,7.82155l0.867949,1.218704l7.26506,10.166271h-2.981339L11.436522,13.338465z",
      },
      "type": "path",
    },
  ],
  "props": {
    "className": undefined,
    "viewBox": "0 0 24 24",
  },
  "type": "svg",
}
`;

exports[`Icon renders correctly with custom size 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "x",
      ],
      "props": {},
      "type": "title",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "d": "M14.095479,10.316482L22.286354,1h-1.940718l-7.115352,8.087682L7.551414,1H1l8.589488,12.231093L1,23h1.940717  l7.509372-8.542861L16.448587,23H23L14.095479,10.316482z M11.436522,13.338465l-0.871624-1.218704l-6.924311-9.68815h2.981339  l5.58978,7.82155l0.867949,1.218704l7.26506,10.166271h-2.981339L11.436522,13.338465z",
      },
      "type": "path",
    },
  ],
  "props": {
    "className": undefined,
    "viewBox": "0 0 24 24",
  },
  "type": "svg",
}
`;

exports[`Icon renders correctly with custom color 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "x",
      ],
      "props": {},
      "type": "title",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "d": "M14.095479,10.316482L22.286354,1h-1.940718l-7.115352,8.087682L7.551414,1H1l8.589488,12.231093L1,23h1.940717  l7.509372-8.542861L16.448587,23H23L14.095479,10.316482z M11.436522,13.338465l-0.871624-1.218704l-6.924311-9.68815h2.981339  l5.58978,7.82155l0.867949,1.218704l7.26506,10.166271h-2.981339L11.436522,13.338465z",
      },
      "type": "path",
    },
  ],
  "props": {
    "className": undefined,
    "viewBox": "0 0 24 24",
  },
  "type": "svg",
}
`;

exports[`Icon renders correctly with custom size and color 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "x",
      ],
      "props": {},
      "type": "title",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "d": "M14.095479,10.316482L22.286354,1h-1.940718l-7.115352,8.087682L7.551414,1H1l8.589488,12.231093L1,23h1.940717  l7.509372-8.542861L16.448587,23H23L14.095479,10.316482z M11.436522,13.338465l-0.871624-1.218704l-6.924311-9.68815h2.981339  l5.58978,7.82155l0.867949,1.218704l7.26506,10.166271h-2.981339L11.436522,13.338465z",
      },
      "type": "path",
    },
  ],
  "props": {
    "className": undefined,
    "viewBox": "0 0 24 24",
  },
  "type": "svg",
}
`;
